robots/prowjob-experiment/handler/handler.go

```diff
@@ -0,0 +1,320 @@
+	}
+
+	// TODO: handle changes to config.yaml and presets
+	modifiedJobConfigs, err := getModifiedConfigs(repo, event, jobConfigPatterns)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not get the modified configs")
+		return
+	}
+
+	if len(modifiedJobConfigs) == 0 {
+		logrus.Infoln("no job configs were modified - nothing to do")
+		return
+	}
+
+	modifiedConfigs, err := loadConfigsAtRef(repo, event.PullRequest.Head.SHA, prowConfigPath, modifiedJobConfigs)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not load the modified configs from pull request's head")
+		return
+	}
+
+	originalConfigs, err := loadConfigsAtRef(repo, event.PullRequest.Base.SHA, prowConfigPath, modifiedJobConfigs)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not load the modified configs from pull request's base")
+		return
+	}
+
+	squashedConfigs := squashConfigPaths(originalConfigs, modifiedConfigs)
+
+	prowJobs := generateProwJobs(squashedConfigs, event)
+	createProwJobs(cl, prowJobs)
+}
+
+// Creates prow jobs
+func createProwJobs(cl *kube.Client, jobs []prowapi.ProwJob) {
+	for _, job := range jobs {
+		_, err := cl.CreateProwJob(job)
+		if err != nil {
+			logrus.WithError(err).Fatal("failed to create new ProwJob")
+		}
+	}
+}
+
+// Get the modified job configs from the repo
+func getModifiedConfigs(repo *git.Repo, event *github.PullRequestEvent, pattern string) ([]string, error) {
+	changes, err := repo.Diff(event.PullRequest.Head.SHA, event.PullRequest.Base.SHA)
+	if err != nil {
+		return nil, err
+	}
+	return filterByPattern(changes, pattern)
+}
+
+// Filter the input array/slice by the given pattern.
+// The pattern has to be a shell file name pattern.
+// https://golang.org/pkg/path/filepath/#Match
+func filterByPattern(input []string, pattern string) ([]string, error) {
+	var out []string
+	for _, s := range input {
+		match, err := filepath.Match(pattern, s)
+		if err != nil {
+			// The only possible error is ErrorBadPattern
+			return nil, err
+		}
+		if match {
+			out = append(out, s)
+		}
+	}
+	return out, nil
+}
+
+// Determine if we should perform on the given event.
+func shouldPerformOnEventAction(event *github.PullRequestEvent) bool {
+	// See https://developer.github.com/v3/activity/events/types/#pullrequestevent
+	// for details
+	switch event.Action {
+	case "opened":
+		return true
+	case "edited":
+		return true
+	case "synchronize":
+		return true
+	default:
+		return false
+	}
+}
+
+// Load job configs but checkout before
+func loadConfigsAtRef(
+	repo *git.Repo, ref, prowConfigPath string, jobConfigPaths []string) (jobConfigPath, error) {
+	if err := repo.Checkout(ref); err != nil {
+		return nil, err
+	}
+
+	return loadConfigs(repo.Directory(), prowConfigPath, jobConfigPaths), nil
+}
+
+// Squash original and modified configs at the same path, returning an array of configs
+// that contain only new and modified job configs.
+func squashConfigPaths(originalConfigs, modifiedConfigs jobConfigPath) []*config.Config {
```

Comment:
This method name sounds like there is more to come here? Is that intended? I mean, wouldn't we want to separate other things into other robots? We could share code in a common base though...